pyro/distributions/normal.py

```diff
@@ -105,3 +108,61 @@ def analytic_var(self):
         Ref: :py:meth:`pyro.distributions.distribution.Distribution.analytic_var`
         """
         return torch.pow(self.sigma, 2)
+
+
+class TorchNormal(Distribution):
+    """
+    Compatibility wrapper around
+    `torch.distributions.Normal <http://pytorch.org/docs/master/_modules/torch/distributions.html#Normal`_
+    """
+    try:
+        reparameterized = torch.distributions.Normal.reparameterized
+    except (ImportError, AttributeError):
+        reparameterized = False
+    enumerable = False
+
+    def __init__(self, mu, sigma, *args, **kwargs):
+        self._torch_dist = torch.distributions.Normal(mean=mu, std=sigma)
+        self._param_shape = torch.Size(broadcast_shape(mu.size(), sigma.size(), strict=True))
+        super(TorchNormal, self).__init__(*args, **kwargs)
+
+    def batch_shape(self, x=None):
+        x_shape = [] if x is None else x.size()
+        shape = torch.Size(broadcast_shape(x_shape, self._param_shape, strict=True))
```

Comment:
This might lead to some hard to find bugs like #414 if the event dimensions do not match between the data and the parameters. Should we allow this kind of broadcasting, or limit it to the batch dimensions only (i.e. any `x`'s rightmost sizes must exactly agree with `sample_shape + event_shape`)?

Reply:
Right, that was actually needed to pass some of the expected-error tests.